### ISSUE TITLE
To Be Done

### ISSUE THREAD

**Issue by ashish2199** [ashish2199/Aidos#6]
I have uploaded initial code for the repository.
I am using Netbeans 8.2 with JDK 1.8. 

I have implemented a animation for walking in the downward direction.
Here are things we need to do
1. Try to figure out if there could be a better for making this animation. The approach I have used now feels very length.
2. Implement animation for walking in left , right and up direction.
3. Registering keyboard inputs

## Let me know if you are having trouble in running the code.

**Comment by llllllxu** [comment]
Does anyone use eclipse? There are lots of errors when I try to import it.

**Comment by lyx13710** [comment]
@llllllxu It is supposed to be a netbean project. @Ashish2199 Maybe you
should convert this project to maven?

**Comment by ashish2199** [comment]
Okay, I will try to use as a maven project.

**Comment by dostertag** [comment]
Just a note.  I'm using netbeans as well and had errors when I initially fetched the repo.  I had to manually select the 1.8 jdk which corrected all errors.

**Comment by ashish2199** [comment]
@llllllxu 
It maybe that you are using a older JDK or eclipse version. 

We will be using JavaFX 8 which is a part of JDK 8.

I cloned the project and opened in it Eclipse Oxygen and didn't get any errors.   
![Screenshot](https://user-images.githubusercontent.com/7945202/32898942-94c7df14-cb0f-11e7-959c-08f4bc481d6d.png)

Did anyone else get error on opening the project ?

**Comment by lyx13710** [comment]
@ashish2199 I got no serious error in Intellij. Can run the program fine  without any modification. Btw, shouldn't we open a new issue for this one?

**Comment by CiaranRoft** [comment]
Got it up and running in Netbeans, no major problems!

**Comment by ashish2199** [comment]
@lyx13710   Yeah we should. It will help others who face similar issues.

**Comment by HarrisJT** [comment]
I suggest Google style which can be found on github and imported into editors to autoformat.

**Comment by lyx13710** [comment]
I found a stackoverflow Q&A which I think is great to implement animation. I'm posting the link here in case you guys are interested. Maybe I'll try to implement it tomorrow.
https://gamedev.stackexchange.com/questions/53705/how-can-i-make-a-sprite-sheet-based-animation-system

**Comment by llllllxu** [comment]
Thank you guys. It works now.

**Comment by dostertag** [comment]
Bear with me, guys.  I'm very new to Java and am really glad Ashish created this project.  That said, I'm having difficulty wrapping my head around the Entity class and how the other player/ai/stationary objects will inherit from that interface.  Any chance some of you have experience creating an UML to explain?  
From my (limited) understanding, I see the Entity interface as having:
a position(x, y)
boolean moveable
health
image/sprite

Player would implement Entity and add
void move(event e)???
animate(Direction direction)
attack(dropBomb?)

I realize this is basic and incomplete...  I'm willing to learn how to create an UML with easyUML1(netbeans plugin) if no one knows how to but will need a more fleshed-out direction to start.  

It seems to me that we'll need to have a basic 'player' implementation before we can set the movement and direction efficiently.  

Again, I'm a beginner, and am happy that this was posted in the first place.

**Comment by CoreyHendrey** [comment]
Proposed a change the animation system, let me know what you guys think. :)

**Comment by lyx13710** [comment]
@CoreyHendrey Having a separate sprite model and animator class might be a good idea. I was already planning to do that last night.

**Comment by ashish2199** [comment]
@lyx13710  I didn't understand as to what exactly you refer to by that.
If you wanted to ask my view on changes proposed by Corey, I feel he has done a lot of things. I am at present analyzing them as to whether the approach taken is good or not in the long run.

**Comment by lyx13710** [comment]
@ashish2199 Sorry. I'm having trouble expressing my thoughts right now. I think I asked whether you have any new codes/changes. Urgg, I'm deleting these stupid comments. Sorry.

**Comment by StewartWallace1115** [comment]
What class should I put character movement? In the Direction class?

**Comment by CoreyHendrey** [comment]
Depends. If Ashish accepts my PR it’ll probably be in CharacterEntity.

**Comment by ashish2199** [comment]
@StewartWallace1115  what do you refer by character movement ? key events handling or movement of player that is the animation that player will use to walk.   I didnt understand.
If anyone else understood please clarify.
